### ISSUE TITLE
Crash with PostgreSQL 10 in DeconstructQualifiedName

### ISSUE THREAD

**Issue by df7cb** [pllua/pllua-deprecated#55]
Hi,
running sql/plluatest.sql against PostgreSQL crashes the server:
```
# \i sql/plluatest.sql 
CREATE EXTENSION
...
SELECT * FROM get_rows('name');

Program received signal SIGSEGV, Segmentation fault.
DeconstructQualifiedName (names=0x4ca80e30, 
    nspname_p=nspname_p@entry=0x7ffecf6ba9e0, 
    objname_p=objname_p@entry=0x7ffecf6ba9e8)
    at ./build/../src/backend/catalog/namespace.c:2777
2777	./build/../src/backend/catalog/namespace.c: Datei oder Verzeichnis nicht gefunden.
(gdb) bt f
#0  DeconstructQualifiedName (names=0x4ca80e30, 
    nspname_p=nspname_p@entry=0x7ffecf6ba9e0, 
    objname_p=objname_p@entry=0x7ffecf6ba9e8)
    at ./build/../src/backend/catalog/namespace.c:2777
        catalogname = <optimized out>
        schemaname = 0x0
        objname = 0x0
        __func__ = "DeconstructQualifiedName"
#1  0x000055824b44903d in LookupTypeName (pstate=pstate@entry=0x0, 
    typeName=typeName@entry=0x55824ca80e68, typmod_p=typmod_p@entry=0x0, 
    missing_ok=missing_ok@entry=0 '\000')
    at ./build/../src/backend/parser/parse_type.c:152
        schemaname = 0xd9 <error: Cannot access memory at address 0xd9>
        typname = 0x55824ca80d40 "\177\177\177\177\177\177\177\177\b"
        typoid = <optimized out>
        tup = <optimized out>
        __func__ = "LookupTypeName"
#2  0x000055824b449779 in typenameType (pstate=pstate@entry=0x0, 
    typeName=0x55824ca80e68, typmod_p=typmod_p@entry=0x0)
    at ./build/../src/backend/parser/parse_type.c:251
        tup = <optimized out>
        __func__ = "typenameType"
#3  0x00007f1238ee9778 in luaP_gettypeoid (
    type_name=type_name@entry=0x55824ca1f118 "text")
    at /srv/projects/postgresql/pllua/postgresql-pllua/plluaspi.c:968
        namelist = 0x4ca80e30
        typetup = <optimized out>
        typeoid = <optimized out>
#4  0x00007f1238ee9a74 in luaP_prepare (L=0x55824ca75f30)
    at /srv/projects/postgresql/pllua/postgresql-pllua/plluaspi.c:998
        s = 0x55824ca1f118 "text"
        type = <optimized out>
        k = <optimized out>
        nargs = 1
        q = 0x55824ca75c18 "SELECT sid, sname, sdata FROM sometable WHERE sname = $1"
        p = 0x55824ca1f1c8
#5  0x00007f1238cb93b0 in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.1.so.0
No symbol table info available.
#6  0x00007f1238cc4062 in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.1.so.0
No symbol table info available.
#7  0x00007f1238cb8ade in ?? () from /usr/lib/x86_64-linux-gnu/liblua5.1.so.0
No symbol table info available.
#8  0x00007f1238cb98bf in lua_resume ()
   from /usr/lib/x86_64-linux-gnu/liblua5.1.so.0
No symbol table info available.
#9  0x00007f1238ee5c83 in luaP_callhandler (L=0x55824ca5d100, 
    fcinfo=0x7ffecf6bb090)
    at /srv/projects/postgresql/pllua/postgresql-pllua/plluaapi.c:1346
        status = <optimized out>
---Type <return> to continue, or q <return> to quit---
        hasresult = <optimized out>
        rsi = 0x7ffecf6bb050
        save_exception_stack = 0x7ffecf6bb660
        save_context_stack = 0x0
        local_sigjmp_buf = {{__jmpbuf = {94018119789720, -892343347548305222, 1, 
              0, 0, 1, -892343347579762502, -989615990724278086}, 
            __mask_was_saved = 0, __saved_mask = {__val = {0, 0, 0, 0, 
                139716539865008, 139716539865008, 94018119860768, 140732378361632, 
                94018100259894, 139716539865008, 94018119873384, 140732378361664, 
                94018099960385, 140732378361712, 94018097515833, 
                140732378361696}}}}
        retval = 0
        base = 0
        fi = 0x55824ca1ec98
        prev = 0x0
        istrigger = 0 '\000'
        __func__ = "luaP_callhandler"
#10 0x000055824b4e14c4 in ExecMakeTableFunctionResult (setexpr=0x55824ca7b338, 
    econtext=0x55824ca7b030, argContext=<optimized out>, 
    expectedDesc=0x55824ca7c278, randomAccess=0 '\000')
    at ./build/../src/backend/executor/execSRF.c:231
        result = 94018119799464
        tupstore = 0x0
        tupdesc = 0x0
        funcrettype = 32806
        returnsTuple = <optimized out>
        returnsSet = 1 '\001'
        fcinfo = {flinfo = 0x55824ca7b358, context = 0x0, 
          resultinfo = 0x7ffecf6bb050, fncollation = 100, isnull = 0 '\000', 
          nargs = 1, arg = {94018119507120, 140732378362080, 0, 31285314337466112, 
            1125891316908032, 0, 94018099506784, 281483566645432, 2, 0, 
            94018119169872, 0, 32772, 0 <repeats 18 times>, 94018100112036, 0, 0, 
            4, 0, 94018119008256, 139716539834160, 139716539834160, 
            94018119860768, 140732378362384, 5363547273144133376, 139716539834160, 
            1, 140732378362368, 94018099953837, 94018119348072, 139716540415600, 
            139716540415600, 2409541699, 140732378362864, 94018099960223, 
            140732378316800, 140732378362511, 14170108208, 140732378362528, 
            140732378362512, 5363547273144133376, 140732378362928, 1, 
            140732378362560, 94018099953837, 94018119348072, 139716540415600, 
            139716540415600, 2409541699, 140732378362992, 94018099960223, 25, 
            140732378362639, 12884901888, 140732378362656, 140732378362640, 
            140732378362656, 0, 0, 1125891316908032, 0, 94018099506784, 
            281483566645432, 2, 0, 94018119169872, 0, 25, 0 <repeats 15 times>}, 
          argnull = '\000' <repeats 40 times>, "g\354QK\202U\000\000\000\000\000\000\000\000\000\000\020ͧL\202U\000\000(ǧL\202U\000\000\031\000\000\000\000\000\000\000\003", '\000' <repeats 15 times>, "p\264", <incomplete sequence \317>}
        fcusage = {fs = 0x0, save_f_total_time = {tv_sec = 94018098002939, 
            tv_nsec = 94018119169872}, save_total = {tv_sec = 5363547273144133376, 
            tv_nsec = 94018098002416}, f_start = {tv_sec = 5363547273144133376, 
            tv_nsec = 140732378361984}}
---Type <return> to continue, or q <return> to quit---
        rsinfo = {type = T_ReturnSetInfo, econtext = 0x55824ca7b030, 
          expectedDesc = 0x55824ca7c278, allowedModes = 11, 
          returnMode = SFRM_ValuePerCall, isDone = ExprSingleResult, 
          setResult = 0x0, setDesc = 0x0}
        tmptup = {t_len = 3479941568, t_self = {ip_blkid = {bi_hi = 32766, 
              bi_lo = 0}, ip_posid = 4864}, t_tableOid = 21890, t_data = 0x0}
        callerContext = 0x55824ca33818
        first_time = 1 '\001'
        __func__ = "ExecMakeTableFunctionResult"
#11 0x000055824b4eea07 in FunctionNext (node=node@entry=0x55824ca7af18)
    at ./build/../src/backend/executor/nodeFunctionscan.c:94
        tstore = 0x0
        estate = <optimized out>
        direction = ForwardScanDirection
        scanslot = 0x55824ca7b220
        alldone = <optimized out>
        oldpos = <optimized out>
        funcno = <optimized out>
        att = <optimized out>
#12 0x000055824b4e03ea in ExecScanFetch (
    recheckMtd=0x55824b4ee760 <FunctionRecheck>, 
    accessMtd=0x55824b4ee770 <FunctionNext>, node=0x55824ca7af18)
    at ./build/../src/backend/executor/execScan.c:95
        estate = <optimized out>
#13 ExecScan (node=0x55824ca7af18, accessMtd=0x55824b4ee770 <FunctionNext>, 
    recheckMtd=0x55824b4ee760 <FunctionRecheck>)
    at ./build/../src/backend/executor/execScan.c:143
        econtext = 0x55824ca7b030
        qual = 0x0
        projInfo = 0x0
#14 0x000055824b4d9d2d in ExecProcNode (node=0x55824ca7af18)
    at ./build/../src/include/executor/executor.h:250
No locals.
#15 ExecutePlan (execute_once=<optimized out>, dest=0x55824ca76de8, 
    direction=<optimized out>, numberTuples=0, sendTuples=<optimized out>, 
    operation=CMD_SELECT, use_parallel_mode=<optimized out>, 
    planstate=0x55824ca7af18, estate=0x55824ca7ad00)
    at ./build/../src/backend/executor/execMain.c:1720
        slot = <optimized out>
        current_tuple_count = 0
#16 standard_ExecutorRun (queryDesc=0x55824ca42650, direction=<optimized out>, 
    count=0, execute_once=<optimized out>)
    at ./build/../src/backend/executor/execMain.c:363
        estate = 0x55824ca7ad00
        operation = CMD_SELECT
        dest = 0x55824ca76de8
        sendTuples = <optimized out>
        __func__ = "standard_ExecutorRun"
#17 0x000055824b62fe06 in PortalRunSelect (portal=portal@entry=0x55824ca2e210, 
    forward=forward@entry=1 '\001', count=0, count@entry=9223372036854775807, 
---Type <return> to continue, or q <return> to quit---
    dest=dest@entry=0x55824ca76de8) at ./build/../src/backend/tcop/pquery.c:932
        queryDesc = 0x55824ca42650
        direction = <optimized out>
        nprocessed = <optimized out>
        __func__ = "PortalRunSelect"
#18 0x000055824b631670 in PortalRun (portal=portal@entry=0x55824ca2e210, 
    count=count@entry=9223372036854775807, isTopLevel=isTopLevel@entry=1 '\001', 
    run_once=run_once@entry=1 '\001', dest=dest@entry=0x55824ca76de8, 
    altdest=altdest@entry=0x55824ca76de8, completionTag=0x7ffecf6bb7b0 "")
    at ./build/../src/backend/tcop/pquery.c:773
        save_exception_stack = 0x7ffecf6bb9e0
        save_context_stack = 0x0
        local_sigjmp_buf = {{__jmpbuf = {0, -892343346740901702, 94018119852560, 
              140732378363824, 94018119852560, 94018119504720, 
              -892343346826884934, -6384755834651719494}, __mask_was_saved = 0, 
            __saved_mask = {__val = {94016834109440, 1, 0, 140731715569261, 6, 6, 
                140732378363822, 1, 88, 94018119852560, 2, 94018120150448, 
                94018119504720, 140732378363696, 94018100243402, 2}}}}
        result = <optimized out>
        nprocessed = <optimized out>
        saveTopTransactionResourceOwner = 0x55824c987de0
        saveTopTransactionContext = 0x55824ca32aa8
        saveActivePortal = 0x0
        saveResourceOwner = 0x55824c987de0
        savePortalContext = 0x0
        saveMemoryContext = 0x55824ca32aa8
        __func__ = "PortalRun"
#19 0x000055824b62cd40 in exec_simple_query (
    query_string=0x55824c9d82b0 "SELECT * FROM get_rows('name');")
    at ./build/../src/backend/tcop/postgres.c:1099
        portal = 0x55824ca2e210
        snapshot_set = <optimized out>
        commandTag = <optimized out>
        completionTag = "\000\000\000\000\002\000\000\000\360\267k\317\376\177\000\000V\215w\317\376\177", '\000' <repeats 11 times>, "\270k\317\001\000\000\000\260\202\235L\202U\000\000\260\202\235L\202U\000\000(F\230L\202U\000"
        querytree_list = <optimized out>
        plantree_list = <optimized out>
        receiver = 0x55824ca76de8
        format = 0
        dest = DestRemote
        parsetree_list = 0x55824c9d9378
        parsetree_item = 0x55824c9d9350
        save_log_statement_stats = 0 '\000'
        was_logged = 0 '\000'
        isTopLevel = 1 '\001'
        msec_str = "\000\000\000\000\002\000\000\000\360\267k\317\376\177\000\000V\215w\317\376\177\000\000\000\000\000\000\000\000\000"
        __func__ = "exec_simple_query"
#20 0x000055824b62ecdf in PostgresMain (argc=<optimized out>, 
---Type <return> to continue, or q <return> to quit---
    argv=argv@entry=0x55824c9847e8, dbname=<optimized out>, 
    username=<optimized out>) at ./build/../src/backend/tcop/postgres.c:4090
        query_string = 0x55824c9d82b0 "SELECT * FROM get_rows('name');"
        input_message = {data = 0x55824c9d82b0 "SELECT * FROM get_rows('name');", 
          len = 32, maxlen = 1024, cursor = 32}
        local_sigjmp_buf = {{__jmpbuf = {140732378364288, -892343346898188102, 1, 
              94018119157288, 1, 94018105224320, -892343346975782726, 
              -6384755828189738822}, __mask_was_saved = 1, __saved_mask = {
              __val = {0, 0, 5363547273144133376, 8192, 94018101829342, 
                140732378364816, 140732378365216, 140732378366256, 0, 0, 
                139716497385992, 75, 206158430256, 140732378364792, 
                140732378364576, 94018119157456}}}}
        send_ready_for_query = 0 '\000'
        disable_idle_in_transaction_timeout = 0 '\000'
        __func__ = "PostgresMain"
#21 0x000055824b319a59 in BackendRun (port=0x55824c97b8a0)
    at ./build/../src/backend/postmaster/postmaster.c:4357
        ac = 1
        secs = 558470513
        usecs = 483173
        i = 1
        av = 0x55824c9847e8
        maxac = <optimized out>
#22 BackendStartup (port=0x55824c97b8a0)
    at ./build/../src/backend/postmaster/postmaster.c:4029
        bn = <optimized out>
        pid = <optimized out>
#23 ServerLoop () at ./build/../src/backend/postmaster/postmaster.c:1753
        i = <optimized out>
        rmask = {fds_bits = {8, 0 <repeats 15 times>}}
        selres = <optimized out>
        now = <optimized out>
        readmask = {fds_bits = {72, 0 <repeats 15 times>}}
        last_lockfile_recheck_time = 1505155294
        last_touch_time = 1505155146
        __func__ = "ServerLoop"
#24 0x000055824b5afb18 in PostmasterMain (argc=5, argv=0x55824c94fe30)
    at ./build/../src/backend/postmaster/postmaster.c:1361
        opt = <optimized out>
        status = <optimized out>
        userDoption = <optimized out>
        listen_addr_saved = <optimized out>
        i = <optimized out>
        output_config_variable = <optimized out>
        __func__ = "PostmasterMain"
#25 0x000055824b31b934 in main (argc=5, argv=0x55824c94fe30)
    at ./build/../src/backend/main/main.c:228
No locals.
```

**Comment by df7cb** [comment]
Any takers? The 10.0 release is getting closer.

**Comment by eugwne** [comment]
I'll try to find some time this week  to fix it. Is it REL_10_RC1 ?

**Comment by df7cb** [comment]
REL_10_RC1 is the most recent tag on the REL_10_STABLE branch, yes.

If you are on Debian/Ubuntu/RH/CentOS, there's PG 10 packages on
apt.postgresql.org resp. yum.postgresql.org.

Thanks!

**Comment by df7cb** [comment]
pllua is among the last 3 packages blocking the transition from PG 9.6 to 10 in Debian. Is there anything I could do to help?

**Comment by eugwne** [comment]
fixed in #56

**Comment by df7cb** [comment]
Packages updated, everything looking good, thanks!
